File: javascript/week2/ex01-contact/contact.test.js
// @ts-nocheck
const { createContact } = require('./contact');

describe('Contact Object', () => {
  test('should create a contact correctly', () => {
    const contact = createContact('Mariane', 'mari@test.com', '123');
    expect(contact.name).toBe('Mariane');
    expect(contact.email).toBe('mari@test.com');
  });

  test('should update email using the method', () => {
    const contact = createContact('Mariane', 'mari@test.com', '123');
    contact.updateEmail('novo@test.com');
    expect(contact.email).toBe('novo@test.com');
  });
});

File: GO/week2/ex01-contact/contact_test.go
package main

import "testing"

func TestContactLogic(t *testing.T) {
	// 1. Testa a criação
	c := CreateContact("Davi", "davi@test.com", "12345")
	if c.Name != "Davi" || c.Email != "davi@test.com" {
		t.Errorf("Falha ao criar contato!")
	}

	// 2. Testa a atualização do email (Ponteiro)
	c.UpdateEmail("davi.novo@test.com")
	if c.Email != "davi.novo@test.com" {
		t.Errorf("O email não foi atualizado corretamente. Recebi: %s", c.Email)
	}
}
